### PR TITLE
Add creation plan

### PR DESCRIPTION
**Pull request by Hiieu**
👋 In this PR, we've added the creation_plan block to allow the user to specify an optimised hardware spec for their databases in the cluster.

Changes:
- Adjust the CRUDI operations for the new block.
- Update the acceptance tests
- Update the docs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,7 +28,11 @@ website/node_modules
 *.iml
 
 website/vendor
+vendor
+.vscode/
 
+# makefile
```
Comment (Hiieu) [inline .gitignore:34]: Do we need to ignore those (lines: 34-35)?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file.
 See updating [Changelog example here](https://keepachangelog.com/en/1.0.0/)
 
+## 1.0.0 (X July 2022)
+
+### Added
+- Added the creation_plan block in the subscription resource schema.
+
```
Comment (greg-oc) [inline CHANGELOG.md:9]: Perhaps we should also mention that the "database" block has been removed here

**Review by greg-oc — COMMENTED** [review]
Looks good, would just like to discuss the comments before approving

File: .gitignore (hunk 1 above)
Comment (greg-oc) [reply .gitignore:34]: Not sure, probably best to remove those lines from the .gitignore

File: internal/provider/resource_rediscloud_subscription.go [hunk 3]
```diff
@@ -227,65 +205,28 @@ func resourceRedisCloudSubscription() *schema.Resource {
...
-			"database": {
-				Description: "A database object",
-				Type:        schema.TypeSet,
-				Required:    true,
-				MinItems:    1,
+			"creation_plan": {
+				Description: "Information about the planned databases used to optimise the database infrastructure. This information is only used when creating a new subscription and any changes will be ignored after this.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				// The block is required when the user provisions a new subscription.
```
[the remark this answers is not in the document]
Comment (Hiieu) [reply internal/provider/resource_rediscloud_subscription.go:212]: Yes, the creation_plan block is a required block.

**Review by greg-oc — APPROVED** [review]
Looks good

**Pull request merged**